### DISCUSSION TITLE
fracAGN mock analysis

### DISCUSSION THREAD

**Discussion by olliepye** [mboquien/cigale#13] [Q&A] [answered]
Hi all,

When I perform a mock analysis on fracAGN (skirtor2016), the results are quite scattered as shown below. Do you have any tips to constrain this? I am using the ZFOURGE catalogue which contains galaxies from $0.2 < z < 4.0$ mostly. I am trying to create a generalised parameter space to determine agn contribution to all galaxies. 

![agn_frac](https://user-images.githubusercontent.com/42687811/173975380-b2c11eaf-56d8-4380-820a-9bf49a308179.png)

Dust Luminosity seems well constrained. AGN luminosity has some scatter and wild points. I have done runs where I have increased the agn frac steps by 0.05. This better constrains AGN luminosity but not agn Frac.

![dust_lum](https://user-images.githubusercontent.com/42687811/173975554-644ce279-ab60-49fe-ac44-503c9f73705a.png)

![agn_lum](https://user-images.githubusercontent.com/42687811/173975559-ba70f181-930e-4c64-bbfd-00c0955f1bb0.png)

**AGN frac cropped:**
![agn_lum_crop](https://user-images.githubusercontent.com/42687811/173975557-beddb6a9-8146-45f3-bbbf-3a282a577a34.png)

My skirtor parameter space is below. 
```
   [[skirtor2016]]
    # Average edge-on optical depth at 9.7 micron; the actual one alongthe
    # line of sight may vary depending on the clumps distribution. Possible
    # values are: 3, 5, 7, 9, and 11.
    t = 7
    # Power-law exponent that sets radial gradient of dust density.Possible
    # values are: 0., 0.5, 1., and 1.5.
    pl = 1.0
    # Index that sets dust density gradient with polar angle.Possible values
    # are:  0., 0.5, 1., and 1.5.
    q = 1.0
    # Angle measured between the equatorial plane and edge of the torus.
    # Half-opening angle of the dust-free (or polar-dust) cone is 90°-oa.
    # Possible values are: 10, 20, 30, 40, 50, 60, 70, and 80
    oa = 40
    # Ratio of outer to inner radius, R_out/R_in.Possible values are: 10,
    # 20, and 30
    R = 20
    # fraction of total dust mass inside clumps. 0.97 means 97% of total
    # mass is inside the clumps and 3% in the interclump dust. Possible
    # values are: 0.97.
    Mcl = 0.97
    # inclination, i.e. viewing angle, position of the instrument w.r.t. the
    # AGN axis. i=[0, 90°-oa): face-on, type 1 view; i=[90°-oa, 90°]: edge-
    # on, type 2 view. Possible values are: 0, 10, 20, 30, 40, 50, 60, 70,
    # 80, and 90.
    i = 30, 70
    # Disk spectrum: 0 for the regular Skirtor spectrum, 1 for the
    # Schartmann (2005) spectrum.
    disk_type = 1
    # Power-law of index δ modifying the optical slop of the disk. Negative
    # values make the slope steeper where as positive values make it
    # shallower.
    delta = -0.36
    # AGN fraction.
    fracAGN = 0, 0.1, 0.2, 0.3, 0.4, 0.5, 0.6, 0.7, 0.8, 0.9, 0.99
    # Wavelength range in microns where to compute the AGN fraction. Note
    # that it includes all the components, not just dust emission. To use
    # the the total dust luminosity set to 0/0.
    lambda_fracAGN = 0/0
    # Extinction law of the polar dust: 0 (SMC), 1 (Calzetti 2000), or 2
    # (Gaskell et al. 2004)
    law = 0
    # E(B-V) for the extinction in the polar direction in magnitudes.
    EBV = 0.0, 0.03, 0.1, 0.2, 0.4, 0.6, 1.0, 1.8
    # Temperature of the polar dust in K.
    temperature = 100.0
    # Emissivity index of the polar dust.
    emissivity = 1.6

```

Thanks, 
Ollie

**Comment by Guang91** [comment, marked as answer]
Good question, Ollie.

It's hard to constrain fracAGN because it's defined at the entire IR wavelength range (3-1000 um) by default. There's lots of degeneracy considering you don't have full coverage for this wide range. 

So, I suggest re-defining fracAGN over a wavelength range that your data have good coverage, if you really want to constrain it. You can do this by changing the lambda_fracAGN parameter. 

Thanks,
Guang

**Reply by olliepye** [reply]
Thanks Guang! That makes sense. I'll have a go at changing the lambda_fracAGN parameter.
